core/dbt/parser/sources.py

```diff
@@ -131,32 +131,6 @@ def parse_source(self, target: UnpatchedSourceDefinition) -> SourceDefinition:
         description = table.description or ""
         source_description = source.description or ""
 
-        # We need to be able to tell the difference between explicitly setting the loaded_at_field to None/null
-        # and when it's simply not set.  This allows a user to override the source level loaded_at_field so that
-        # specific table can default to metadata-based freshness.
-        if table.loaded_at_field_present and table.loaded_at_query:
-            raise ParsingError(
-                "Cannot specify both loaded_at_field and loaded_at_query at table level."
-            )
-        if source.loaded_at_field and source.loaded_at_query:
-            raise ParsingError(
-                "Cannot specify both loaded_at_field and loaded_at_query at source level."
-            )
-
-        if table.loaded_at_field_present or table.loaded_at_field is not None:
-            loaded_at_field = table.loaded_at_field
-        else:
-            loaded_at_field = source.loaded_at_field  # may be None, that's okay
-
-        loaded_at_query: Optional[str]
-        if table.loaded_at_query is not None:
-            loaded_at_query = table.loaded_at_query
-        else:
-            if table.loaded_at_field_present:
-                loaded_at_query = None
-            else:
-                loaded_at_query = source.loaded_at_query
```

Comment:
@QMalcolm - FYI I've kept this functionality 1:1 to keep the risk of this change as low as possible.